cargo-dist/templates/ci/github_ci.yml.j2

```diff
@@ -424,10 +427,12 @@ jobs:
           cat dist-manifest.json
           echo "manifest=$(jq -c "." dist-manifest.json)" >> "$GITHUB_OUTPUT"
       - name: "Upload dist-manifest.json"
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          name: artifacts
+          # Overwrite the previous copy
+          name: artifacts-dist-manifest
           path: dist-manifest.json
+          overwrite: true
```

Comment:
wait what the fuck, there's an overwrite flag? god damnit github why was your announcement so scary when the API is way more usable. but anyway, i think it's probably safer for us to actually make plan and host use different artifacts/names. there's no situation where we want "plan OR host". either you definitely want plan, or definitely want host, yeah..? (happy to be corrected here)

Reply:
tbc on my proposal: give host the glorious "dist-manifest.json" name, and downgrade plan to "plan-dist-manifest.json"

Reply:
I feel like the issue is, if they have two different names, we might end up with the wrong one landing in the final release after we merge artifacts later?

I think the thing with `overwrite` is that it doesn't mutate the artifact: it replaces the old artifact with a *new* artifact with the same name. So we can't just treat it like we treated the old mutable artifacts.

Reply:
that's why i'm suggesting the plan one be renamed -- so it *can't* end up being used (github releases rm's all prefixed dist-manifests, and uploads the one non-prefixed one).

Reply:
I missed at first you were talking about prefixing the *manifest*. That's a good idea; pushed a commit which does that, and it's working happily. 

https://github.com/mistydemeo/cargodisttest/releases/tag/v0.2.95

https://github.com/mistydemeo/cargodisttest/actions/runs/7733291447